Importing azure-kusto-data into a TypeScript project fails to build

When I import the latest version of azure-kusto-data into my project, my project no longer builds with TypeScript, and I get these build errors:
```
node_modules/azure-kusto-data/source/azLoginIdentityClient.ts:5:1 - error TS6133: 'TokenResponse' is declared but its value is never read.

5 import {TokenResponse} from "adal-node";
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/azure-kusto-data/source/client.ts:5:18 - error TS7016: Could not find a declaration file for module 'uuid'. 'D:/oss/kusto-test/node_modules/uuid/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/uuid` if it exists or add a new declaration (.d.ts) file containing `declare module 'uuid';`

5 import uuid from "uuid";
                   ~~~~~~

node_modules/azure-kusto-data/source/client.ts:10:17 - error TS2732: Cannot find module '../package.json'. Consider using '--resolveJsonModule' to import module with '.json' extension.

10 import pkg from "../package.json";
                   ~~~~~~~~~~~~~~~~~

Found 3 errors.
```
The issue is the npm package structure includes the JS and TS Declaration files (`.d.ts`) alongside the original source. This is a non-standard project setup and results in imports from the declaration files resolving to the source files. Deleting the source files from the npm package under node_modules fixes the issue.

## Minimal Repro Steps
Create a repo with the following structure:
**package.json**
```json
{
  "name": "kusto-test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "build": "tsc"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "azure-kusto-data": "^2.1.2"
  },
  "devDependencies": {
    "typescript": "^4.1.5"
  }
}
```
**tsconfig.json**
```json
{
    "compilerOptions": {
      "target": "es6",
      "module": "commonjs",
      "moduleResolution": "node",
      "strict": true,
      "types": ["node"],
      "allowSyntheticDefaultImports": true,
      "noUnusedParameters": true,
      "noUnusedLocals": true,
    }
  }

```
**index.ts**
```ts
import 'azure-kusto-data';
```
Then run:
```
npm install
npm run build
```
And observe the above compiler errors.

### Suggested fix
Move all the typescript files into a `src` directory, set the `outDir` to `lib`, and add the `src` to `.npmignore`. Also set the `main` and `typings` fields in package.json to:
```
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
```

Happy to send out a PR if this seems like the right approach. 

@AsafMah since this is related to #102

--- a/azure-kusto-data/.npmignore
+++ b/azure-kusto-data/.npmignore
@@ -1,1 +1,4 @@
 test/*
+# for https://github.com/Azure/azure-kusto-node/issues/115, https://github.com/Microsoft/TypeScript/issues/4667
+source/**/*.ts
+!source/**/*.d.ts

--- a/azure-kusto-data/package.json
+++ b/azure-kusto-data/package.json
@@ -1,6 +1,6 @@
 {
   "name": "azure-kusto-data",
-  "version": "2.1.4",
+  "version": "2.1.5",
   "description": "Azure Data Explorer Query SDK",
   "main": "index.js",
   "types": "index.d.ts",
@@ -29,6 +29,9 @@
   "homepage": "https://github.com/Azure/azure-kusto-node/blob/master/azure-kusto-data/README.md",
   "dependencies": {
     "@azure/ms-rest-nodeauth": "^3.0.3",
+    "@types/mocha": "^8.2.0",
+    "@types/node": "^14.14.13",
+    "@types/uuid": "^8.3.0",
     "adal-node": "^0.1.28",
     "axios": "^0.21.1",
     "moment": "^2.25.3",
@@ -39,9 +42,6 @@
     "sinon": "^7.2.3",
     "tslint": "^6.1.3",
     "typescript": "^4.1.3",
-    "ts-node": "^9.1.1",
-    "@types/mocha": "^8.2.0",
-    "@types/node": "^14.14.13",
-    "@types/uuid": "^8.3.0"
+    "ts-node": "^9.1.1"
   }
 }

--- a/azure-kusto-data/source/azLoginIdentityClient.ts
+++ b/azure-kusto-data/source/azLoginIdentityClient.ts
@@ -2,7 +2,6 @@
 // Licensed under the MIT License.
 
 import {AzureCliCredentials} from "@azure/ms-rest-nodeauth";
-import {TokenResponse} from "adal-node";
 
 export default function acquireToken<T>(connectionString: string, callback: (err: Error | null, data?: { tokenType: string; accessToken: string }) => T) {
 

--- a/azure-kusto-data/test/clientTest.ts
+++ b/azure-kusto-data/test/clientTest.ts
@@ -11,7 +11,6 @@ import {KustoClient} from "../source/client";
 
 import {ClientRequestProperties} from "../source/clientRequestProperties";
 import {
-    KustoResponseDataSet,
     KustoResponseDataSetV1,
     KustoResponseDataSetV2
 } from "../source/response";
@@ -84,7 +83,7 @@ describe("KustoClient", function () {
             const url = "https://cluster.kusto.windows.net";
             const client = new KustoClient(url);
             try{
-                const response = client._parseResponse({}, ExecutionType.Query);
+                client._parseResponse({}, ExecutionType.Query);
             }
             catch(ex){
                 ex.message.startsWith("Failed to parse response ({undefined}) with the following error [TypeError: data.forEach is not a function].");
@@ -98,7 +97,7 @@ describe("KustoClient", function () {
             const client = new KustoClient(url);
 
             try{
-                const response = client._parseResponse(v2ResponseError, ExecutionType.Query);
+                client._parseResponse(v2ResponseError, ExecutionType.Query);
             }
             catch(ex){
                 assert.equal(ex.message.startsWith("Kusto request had errors"), true);
@@ -115,7 +114,7 @@ describe("KustoClient", function () {
             const timeoutMs = moment.duration(2.51, "minutes").asMilliseconds();
             clientRequestProps.setTimeout(timeoutMs);
             client.aadHelper._getAuthHeader = () => { return Promise.resolve("MockToken") };
-            client._doRequest = (endpoint, executionType, headers, payload, timeout, properties) => {
+            client._doRequest = (_endpoint, _executionType, _headers, payload, timeout) => {
                 const payloadObj = JSON.parse(payload);
                 assert.equal(payloadObj.properties.Options.servertimeout, "00:02:30.6");
                 assert.equal(timeout, timeoutMs + moment.duration(0.5, "minutes").asMilliseconds());
@@ -133,8 +132,8 @@ describe("KustoClient", function () {
             const timeoutMs = moment.duration(2.51, "minutes").asMilliseconds();
             clientRequestProps.setClientTimeout(timeoutMs);
             client.aadHelper._getAuthHeader = () => { return Promise.resolve("MockToken") };
-            client._doRequest = (endpoint, executionType, headers, payload, timeout, properties) => {
-                const payloadObj = JSON.parse(payload);
+            client._doRequest = (_endpoint, _executionType, _headers, payload, timeout) => {
+                JSON.parse(payload);
                 assert.equal(timeout, timeoutMs);
                 return Promise.resolve(new KustoResponseDataSetV2([]));
             };
@@ -147,7 +146,7 @@ describe("KustoClient", function () {
             const client = new KustoClient(url);
 
             client.aadHelper._getAuthHeader = () => { return Promise.resolve("MockToken") };
-            client._doRequest = (endpoint, executionType, headers, payload, timeout, properties) => {
+            client._doRequest = (_endpoint, _executionType, _headers, _payload, timeout) => {
                 assert.equal(timeout, moment.duration(4.5, "minutes").asMilliseconds());
                 return Promise.resolve(new KustoResponseDataSetV2([]));
             };
@@ -159,7 +158,7 @@ describe("KustoClient", function () {
             const url = "https://cluster.kusto.windows.net";
             const client = new KustoClient(url);
             client.aadHelper._getAuthHeader = () => { return Promise.resolve("MockToken") };
-            client._doRequest = (endpoint, executionType, headers, payload, timeout, properties) => {
+            client._doRequest = (_endpoint, _executionType, _headers, _payload, timeout) => {
                 assert.equal(timeout, moment.duration(10.5, "minutes").asMilliseconds());
                 return Promise.resolve(new KustoResponseDataSetV2([]));
             };
@@ -175,7 +174,7 @@ describe("KustoClient", function () {
             const clientRequestProps = new ClientRequestProperties();
             clientRequestProps.clientRequestId = clientRequestId;
             client.aadHelper._getAuthHeader = () => { return Promise.resolve("MockToken") };
-            client._doRequest = (endpoint, executionType, headers, payload, timeout, properties) => {
+            client._doRequest = (_endpoint, _executionType, headers) => {
                 assert.equal(headers["x-ms-client-request-id"], clientRequestId);
                 return Promise.resolve(new KustoResponseDataSetV2([]));
             };
@@ -188,7 +187,7 @@ describe("KustoClient", function () {
             const client = new KustoClient(url);
 
             client.aadHelper._getAuthHeader = () => { return Promise.resolve("MockToken") };
-            client._doRequest = (endpoint, executionType, headers, payload, timeout, properties) => {
+            client._doRequest = (endpoint, executionType, _headers, _payload, _timeout, _properties) => {
                 assert.equal(endpoint, `${url}/v1/rest/query`);
                 assert.equal(executionType, ExecutionType.QueryV1);
                 return Promise.resolve(new KustoResponseDataSetV2([]));

--- a/azure-kusto-data/tsconfig.json
+++ b/azure-kusto-data/tsconfig.json
@@ -9,7 +9,9 @@
     "resolveJsonModule": true,
     "allowJs": false,
     "allowSyntheticDefaultImports": true,
-    "declaration": true
+    "declaration": true,
+    "noUnusedParameters": true,
+    "noUnusedLocals": true
   },
   "include": ["source/**/*.ts", "test/**/*.ts", "index.ts"]
 }

--- a/azure-kusto-ingest/.npmignore
+++ b/azure-kusto-ingest/.npmignore
@@ -1,1 +1,4 @@
 test/*
+# for https://github.com/Azure/azure-kusto-node/issues/115, https://github.com/Microsoft/TypeScript/issues/4667
+source/**/*.ts
+!source/**/*.d.ts

--- a/azure-kusto-ingest/index.ts
+++ b/azure-kusto-ingest/index.ts
@@ -9,7 +9,6 @@ import KustoIngestStatusQueues from "./source/status";
 
 import {
     CsvColumnMapping, DataFormat, IngestionMappingType,
-    IngestionProperties,
     JsonColumnMapping,
     ReportLevel,
     ReportMethod,

--- a/azure-kusto-ingest/package.json
+++ b/azure-kusto-ingest/package.json
@@ -1,8 +1,9 @@
 {
   "name": "azure-kusto-ingest",
-  "version": "2.1.4",
+  "version": "2.1.5",
   "description": "Azure Data Explorer Ingestion SDK",
   "main": "index.js",
+  "types": "index.d.ts",
   "engines": {
     "node": ">= 8.0.0"
   },
@@ -35,14 +36,14 @@
     "moment": "^2.22.2",
     "request": "^2.88.0",
     "uuid": "^3.3.2",
-    "uuid-validate": "0.0.3"
-  },
-  "devDependencies": {
+    "uuid-validate": "0.0.3",
     "@types/mocha": "^8.2.0",
     "@types/node": "^14.14.13",
     "@types/sinon": "^9.0.9",
     "@types/uuid": "^8.3.0",
-    "@types/uuid-validate": "0.0.1",
+    "@types/uuid-validate": "0.0.1"
+  },
+  "devDependencies": {
     "mocha": "^7.2.0",
     "sinon": "^7.2.3",
     "ts-node": "^9.1.1",

--- a/azure-kusto-ingest/test/e2eTests/e2eTest.ts
+++ b/azure-kusto-ingest/test/e2eTests/e2eTest.ts
@@ -205,7 +205,7 @@ function main(): void {
         describe('QueryClient', function () {
             it('General BadRequest', async function () {
                 try {
-                    const response = await queryClient.executeQuery(databaseName, "invalidSyntax ");
+                    await queryClient.executeQuery(databaseName, "invalidSyntax ");
                 } catch (ex) {
                     return;
                 }
@@ -214,7 +214,7 @@ function main(): void {
 
             it('PartialQueryFailure', async function () {
                 try {
-                    const response = await queryClient.executeQuery(databaseName, "invalidSyntax ");
+                    await queryClient.executeQuery(databaseName, "invalidSyntax ");
 
                 } catch (ex) {
                     return;

--- a/azure-kusto-ingest/tsconfig.json
+++ b/azure-kusto-ingest/tsconfig.json
@@ -8,7 +8,9 @@
     "sourceMap": true,
     "resolveJsonModule": true,
     "allowJs": false,
-    "declaration": true
+    "declaration": true,
+    "noUnusedParameters": true,
+    "noUnusedLocals": true
   },
   "include": ["source/**/*.ts", "test/**/*.ts", "index.ts"]
 }

